src/Microsoft.Azure.SignalR.Common/Endpoints/ServiceEndpointManagerBase.cs

```diff
@@ -164,31 +164,41 @@ private IReadOnlyList<HubServiceEndpoint> UpdateAndGetRemovedHubServiceEndpoints
             foreach (var hub in _endpointsPerHub.Keys)
             {
                 var oldEndpoints = _endpointsPerHub[hub];
-                var updatedEndpoints = CreateHubServiceEndpoints(hub, endpoints, true);
+                var updatedEndpoints = CreateHubServiceEndpoints(hub, endpoints);
```

Comment:
since it is the removed endpoints, why not select from the existing hub endpoints?